### PR TITLE
Integrated test_flake8.py into CI system

### PR DESCRIPTION
**Pull request by bikramnehra**
This PR should only be merged after https://github.com/kubeflow/kubeflow/pull/2247 is successfully merged.

The PR is to address https://github.com/kubeflow/kubeflow/issues/2388 (integrating test_flake8.py into the CI system).

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/2403)

### CODE REVIEW

File: testing/workflows/components/unit_tests.jsonnet [hunk 1]
```diff
@@ -106,6 +106,26 @@ local dagTemplates = [
     ]),  // create-pr-symlink
     dependencies: ["checkout"],
   },  // create-pr-symlink
+  {
+    template: buildTemplate("flake8-test", [
+      "python",
+      "-m",
+      "testing.test_flake8",
+      "--test_files_dirs=" +
+      srcDir + "/kubeflow/application/tests" + "," +
```
Comment (lluunn) [inline testing/workflows/components/unit_tests.jsonnet]: I think we can just do srcDir + "/kubeflow"? It should recursive walk the tree right?

File: testing/workflows/components/unit_tests.jsonnet [hunk 2]
```diff
@@ -106,6 +106,26 @@ local dagTemplates = [
...
+      srcDir + "/kubeflow/common/tests" + "," +
```
Comment (lluunn) [inline testing/workflows/components/unit_tests.jsonnet]: srcDir + "/testing" should also be added as it has most of the py

**Comment by lluunn** [comment]
/hold
waiting for #2247

File: testing/workflows/components/unit_tests.jsonnet (hunk 2 above)
Comment (bikramnehra) [reply testing/workflows/components/unit_tests.jsonnet]: DONE!

File: testing/workflows/components/unit_tests.jsonnet (hunk 1 above)
Comment (bikramnehra) [reply testing/workflows/components/unit_tests.jsonnet]: After looking at:

https://github.com/kubeflow/kubeflow/pull/2247/commits/2337b2896a093521f8578fde65f2faf7cc515727#diff-e75784592f0b3b405baa4710e7a10ce6R25

I thought there might be a reason to specify sub-directories explicitly.

Yes, it should be a recursive walk. I have implemented the changes.

Also, we might need to propagate these changes here:
https://github.com/kubeflow/kubeflow/blob/master/testing/workflows/components/unit_tests.jsonnet#L116

**Comment by bikramnehra** [comment]
https://github.com/kubeflow/kubeflow/pull/2247 merged
@lluunn can be merged now

**Comment by bikramnehra** [comment]
@lluunn and @jlewi I had a brief look at the failures in the build output of `kubeflow-presubmit` job. It seems like the failures are happening in the tests under `/src/kubeflow/testing/`. 

```
INFO|2019-03-01T17:40:44|/src/kubeflow/testing/py/kubeflow/testing/argo_client.py|24| Workflow kubeflow-presubmit-unittests-2403-c68abe3-6005-d3b9 in namespace kubeflow-test-infra; phase=Failed
```

Since this PR is simply enabling triggering of those tests, therefore we are seeing those failures. IMO a new issue must be filed to fix them.

**Comment by lluunn** [comment]
two test failure:
1. tfjob, I think you just need to merge master
2. flake8. We can't just enable it and let it break presubmit. The PR enable the test should make sure it passes.

**Comment by lluunn** [comment]
thank you!
/lgtm
/approve

**Pull request merged**